app/models/dialog_field_association_validator.rb

```diff
@@ -0,0 +1,19 @@
+class DialogFieldAssociationValidator
+  class DialogFieldAssociationCircularReferenceError < StandardError; end
+
+  def circular_references?(associations)
+    keys_list = associations.collect(&:keys).flatten
+    values_list = associations.collect(&:values).flatten
+    associations.each do |association|
+      association.values.flatten.each do |responder|
+        association.values.first.delete(responder) unless keys_list.include?(responder)
+      end
+    end
+    associations.each do |association|
+      associations.delete(association) unless values_list.include?(association.keys.first)
```

Comment:
Speaking of modifying the associations, I think it's best in a method that we've denoted with the `?` at the end of the method name not to actually do any modifications on arguments that are passed in. It would be really weird if they pass in a list of associations and then get back a different list after asking if it contains circular references, even if currently we're not planning on doing anything with that list. So, whether that means `.dup`/`.clone`-ing `associations` up front or keeping track of the associations to be "deleted" in another separate array is up to you, but `associations` should stay the same.